### PR TITLE
Add new swarm option to attached signed metadata to the swarm's peer entry

### PR DESCRIPTION
**Pull request by Vagabond**

### CODE REVIEW

File: test/peer_SUITE.erl [hunk 1]
```diff
@@ -155,4 +156,19 @@ association_test(_) ->
...
+    #{secret := PrivKey1, public := PubKey1} = libp2p_crypto:generate_keys(ecc_compact),
+    SigFun1 = libp2p_crypto:mk_sig_fun(PrivKey1),
+
+    PeerMap = #{pubkey => libp2p_crypto:pubkey_to_bin(PubKey1),
+                listen_addrs => ["/ip4/8.8.8.8/tcp/1234"],
+                connected => [],
+                nat_type => static,
+                signed_metadata => #{<<"hello">> => <<"world">>}},
+    Peer = libp2p_peer:from_map(PeerMap, SigFun1),
+    ?assertEqual(<<"world">>, libp2p_peer:signed_metadata_get(Peer, <<"hello">>, <<"dlrow">>)),
```
Comment (madninja) [inline test/peer_SUITE.erl]: you're not encoding and decoding the created peer to verify that the proto definition works

File: src/peerbook/libp2p_peer.proto [hunk 2]
```diff
@@ -16,6 +16,7 @@ message peer {
     int64 timestamp = 5;
     map<string, association_list> associations = 6;
     bytes network_id = 7;
+    map<string, bytes> signed_metadata = 8;
```
Comment (madninja) [inline src/peerbook/libp2p_peer.proto]: I see you're using binaries for the keys below. Is this correct?

Comment (Vagabond) [reply src/peerbook/libp2p_peer.proto]: I *think* so? I'm really not sure though?

Comment (madninja) [reply src/peerbook/libp2p_peer.proto]: If you’re using a binary for a key you should change the type in the proto. And add a round trip test

Comment (Vagabond) [reply src/peerbook/libp2p_peer.proto]: I kind of want a string (in a binary) for the key and a richer way to encode the value (so we can have integer/floats/strings/binaries), I'm jut not good at protobuffs

File: test/peer_SUITE.erl (hunk 1 above)
Comment (Vagabond) [reply test/peer_SUITE.erl]: Yeah, I will do that.

File: src/peerbook/libp2p_peer.proto (hunk 2 above)
Comment (madninja) [reply src/peerbook/libp2p_peer.proto]: Pretty sure a string in a binary is still a binary in protobuf

Comment (Vagabond) [reply src/peerbook/libp2p_peer.proto]: You can't use `bytes` as a key, so they're strings forever.

File: test/peer_SUITE.erl (hunk 1 above)
Comment (Vagabond) [reply test/peer_SUITE.erl]: This has been done.

File: src/peerbook/libp2p_peer.erl [hunk 3]
```diff
@@ -368,8 +370,11 @@ decode_unsafe(Bin) ->
 %% throws an error if the peer or one of it's associations can't be
 %% verified
 -spec verify(peer()) -> true.
-verify(Msg=#libp2p_signed_peer_pb{peer=Peer=#libp2p_peer_pb{associations=Assocs}, signature=Signature}) ->
+verify(Msg=#libp2p_signed_peer_pb{peer=Peer0=#libp2p_peer_pb{associations=Assocs, signed_metadata=MD}, signature=Signature}) ->
+    Peer = Peer0#libp2p_peer_pb{signed_metadata=lists:usort(MD)},
+    ct:pal("Verify: Peer ~p", [Peer]),
```
Comment (madninja) [inline src/peerbook/libp2p_peer.erl]: Remove `ct` bits?

**Pull request merged**